api/src/main/java/org/openmrs/module/coreapps/parser/ParseEncounterToJson.java

```diff
@@ -39,6 +39,11 @@ public SimpleObject createEncounterJSON(User authenticatedUser, Encounter encoun
         boolean canDelete = authenticatedUser.hasPrivilege(EmrApiConstants.PRIVILEGE_DELETE_ENCOUNTER);
         boolean canEdit = authenticatedUser.hasPrivilege(EmrApiConstants.PRIVILEGE_EDIT_ENCOUNTER);
 
+        if(encounter.getEncounterType().getEditPrivilege() != null){
```

Comment:
What happens with some on who does not have the EmrApiConstants.PRIVILEGE_EDIT_ENCOUNTER privilege, but has the encounter type edit privilege?